plugins/cachekey/cachekey.h

```diff
@@ -73,6 +73,10 @@ class CacheKey
 private:
   CacheKey(); // disallow
 
+  template <class T>
+  void processHeader(const String &name_s, const ConfigHeaders &config, T &dst,
+                     void (*fun)(const ConfigHeaders &config, const String &name_s, const String &value_s, T &captures));
+
```

Comment:
If this is only a helper for the implementation, maybe we don't need to declare it here?

Reply:
IMHO there is nothing wrong in having helper functions as `private` members of a class. I usually try to have them localized to the unit where they are used but sometimes there are lots of class members used by the function and in this case it feels that the usage would become cumbersome (lots of parameter passing, more chances for mistakes too).